packages/cli/src/executions/legacy-sqlite-execution-recovery.service.ts

```diff
@@ -0,0 +1,58 @@
+import { Logger } from '@n8n/backend-common';
+import { GlobalConfig } from '@n8n/config';
+import { ExecutionData, ExecutionRepository } from '@n8n/db';
+import { Service } from '@n8n/di';
+
+/**
+ * Service for recovering executions that are missing execution data, this should only happen
+ * for sqlite legacy databases.
+ */
+@Service()
+export class LegacySqliteExecutionRecoveryService {
+	private readonly legacySqlite: boolean;
+
+	constructor(
+		private readonly logger: Logger,
+		private readonly executionRepository: ExecutionRepository,
+		private readonly globalConfig: GlobalConfig,
+	) {
+		const { type: dbType, sqlite: sqliteConfig } = this.globalConfig.database;
+		this.legacySqlite = dbType === 'sqlite' && sqliteConfig.poolSize === 0;
+	}
+
+	/**
+	 * Recover key properties of a truncated execution using event logs.
+	 */
+	async cleanupWorkflowExecutions() {
+		if (!this.legacySqlite) {
+			return;
+		}
```

Comment:
Or we can assert in the constructor, so nobody even tries to use this service in any other case.

```ts
assert(isLegacySqlite, 'Only usable when on legacy SQLite driver');
```

Also we should assert that the DB is ready so this service doesn't get called prematurely.